src/main/java/duke/functions/Duke.java

```diff
@@ -0,0 +1,288 @@
+package duke.functions;
+
+import duke.exceptions.EmptyArgException;
+import duke.exceptions.WrongFormatException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Duke {
+
+    private static Boolean isFinished = false;
+    private static final String filepath = "SAVEDLIST.txt";
+
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        ArrayList<Task> items = new ArrayList<>();
+
+        printIntro();
+
+        try {
+            readSavedList(items);
+        } catch (IOException e) {
+            System.out.println("Error creating/reading task list");
+        }
+
+        while (!isFinished) {
+            String userInput = sc.nextLine();
+            String command = userInput.split(" ")[0];
+
+            switch (command) {
+            case "bye":
+                isFinished = true;
+                break;
+            case "list":
+                printTaskList(items);
+                break;
+            case "done": {
+                try {
+                    handleDone(items, userInput);
```

Comment:
nice use of abstractions and SLAP as you have on method "handle"+command for each case...